ios/RNFIRMessaging.m

```diff
@@ -169,7 +257,21 @@ + (void)didReceiveNotificationResponse:(UNNotificationResponse *)response withCo
     if (response.actionIdentifier) {
         [data setValue:response.actionIdentifier forKey:@"_actionIdentifier"];
     }
-    [[NSNotificationCenter defaultCenter] postNotificationName:FCMNotificationReceived object:self userInfo:@{@"data": data, @"completionHandler": completionHandler}];
+
+    if ([response isKindOfClass:UNTextInputNotificationResponse.class]) {
+        [data setValue:[(UNTextInputNotificationResponse *)response userText] forKey:@"_userText"];
+    }
+    
+    NSDictionary *userInfo = @{@"data": data, @"completionHandler": completionHandler};
+
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        if (data[@"_actionIdentifier"] && ![data[@"_actionIdentifier"] isEqualToString:UNNotificationDefaultActionIdentifier]) {
+            initialNotificationActionResponse = userInfo;
```

Comment:
why do you need to assign `initialNotificationActionResponse` here and use it in line 318? can't you just call `postNotificationName`?

Reply:
It's because if the app is killed and the user taps an action, the `didReceiveRemoteNotification:fetchCompletionHandler:` method gets called right after the app starts so the JS context isn't running and handler for that isn't registered. So the notification is kept here and dispatched again after the appropriate handler is registered in `addListener`. I tested it, and it is very robust this way, and it works even if the action isn't marked as [foreground](https://goo.gl/oAUFU9) so the app only have some limited amount of time to do something in the background (30sec I think).

Reply:
thanks.
I figured that yesterday.
I added #792 to handle this issue for all types of notification callbacks
should be compatible with you changes